lang/go/bindings/cne/lport.go

```diff
@@ -0,0 +1,304 @@
+		lportNames = append(lportNames, lportName)
+	}
+
+	sort.Slice(lportNames, func(i, j int) bool { return lportNames[i] < lportNames[j] })
+
+	for _, lportName := range lportNames {
+
+		lportInfo, ok := sys.jcfg.LPortInfoMap()[lportName]
+		if !ok {
+			return fmt.Errorf("LPort %s not found in LPortMap", lportName)
+		}
+
+		if err := sys.setupLPort(lportName, lportInfo); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// cleanupLPorts will cleanup all of the LPorts
+func (sys *System) cleanupLPorts() error {
+
+	for _, lport := range lportList {
+		if lport.lportId < 0xFFFF {
+			ret := C.pktdev_close(C.ushort(lport.lportId))
+			if ret < 0 {
+				return fmt.Errorf("error pktdev_close for lport %s", lport.name)
+			}
+		}
+	}
+	return nil
+}
+
+// LPortMap returns the MAP of LPorts
+func (sys *System) LPortMap() map[string]*LPort {
+
+	return lportMap
+}
+
+// LPortList returns the List of LPorts
+func (sys *System) LPortList() []*LPort {
+
+	return lportList
+}
+
+// LPortByName returns the LPort for the given name
+func (sys *System) LPortByName(name string) *LPort {
+
+	lport, ok := lportMap[name]
+	if !ok {
+		fmt.Printf("LPort %#v not found\n", name)
+		return nil
+	}
+	return lport
+}
+
+// LPortsByName return a list of LPort structure pointers
+func (sys *System) LPortsByName(names []string) []*LPort {
+
+	var lports []*LPort
+
+	for _, name := range names {
+		lport := sys.LPortByName(name)
+		if lport == nil {
+			return nil
+		}
+		lports = append(lports, lport)
+	}
+	return lports
+}
+
+// Name returns the name of the given LPort
+func (p *LPort) Name() string {
+
+	return p.name
+}
+
+// Name returns the name of the given LPort
+func (p *LPort) NetdevName() string {
+
+	return p.netdev
+}
+
+// LPortID returns the lport ID value for the given LPort
+func (p *LPort) LPortID() int {
+
+	return int(p.lportId)
+}
+
+// LPortStats returns the lport statistics structure for the given LPort
+func (p *LPort) LPortStats() (*LPortStats, error) {
+
+	var stats C.lport_stats_t
+
+	if ret := C.pktdev_stats_get(p.lportId, &stats); ret < 0 {
+		return nil, fmt.Errorf("LPortStats failed with error code %d", ret)
+	}
+
+	ps := &LPortStats{}
+	ps.InPackets = uint64(stats.ipackets)
+	ps.InBytes = uint64(stats.ibytes)
+	ps.InErrors = uint64(stats.ierrors)
+	ps.InMissed = uint64(stats.imissed)
+	ps.RxInvalid = uint64(stats.rx_invalid)
+	ps.OutPackets = uint64(stats.opackets)
+	ps.OutBytes = uint64(stats.obytes)
+	ps.OutErrors = uint64(stats.oerrors)
+	ps.OutDropped = uint64(stats.odropped)
+	ps.TxInvalid = uint64(stats.tx_invalid)
+
+	ps.InPacketRate = ps.InPackets - p.prevInPackets
+	ps.OutPacketRate = ps.OutPackets - p.prevOutPackets
+
+	p.prevInPackets = ps.InPackets
+	p.prevOutPackets = ps.OutPackets
+
+	return ps, nil
+}
+
+// ResetLPortStats resets the LPort statistics
+func (p *LPort) ResetLPortStats() error {
+
+	if ret := C.pktdev_stats_reset(p.lportId); ret < 0 {
+		return fmt.Errorf("ResetLPortStats failed with error code %d", ret)
```

Comment:
just again, I'm wondering about log vs fmt...

Reply:
This one we are returning a error object so must be a fmt.Errorf

Reply:
d'oh - of course